python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py

```diff
@@ -269,9 +276,25 @@ def _chat_message_content_to_dict(self, message: "ChatMessageContent") -> dict[s
 
     async def _send_chat_request(self, settings: OpenAIChatPromptExecutionSettings) -> list["ChatMessageContent"]:
         """Send the chat request."""
+        span = model_diagnostics.start_completion_activity(settings.ai_model_id, MODEL_PROVIDER_NAME,
```

Comment:
we should put all this and the bottom part into a single decorator so that we can universally apply this across all connectors!

Reply:
I don't know how feasible or convoluted this can be to achieve as there will be differences between connectors. I am certainly checking right now.

Does this need to be part of this PR, or can it be in a follow-up one?